xls/dslx/dslx_fmt.cc

```diff
@@ -133,38 +136,87 @@ absl::Status RealMain(std::string_view input_path,
     std::cout << formatted << std::flush;
   }
 
-  if (absl::GetFlag(FLAGS_error_on_changes) && formatted != contents) {
+  if (error_on_changes && formatted != contents) {
     return absl::InternalError("Formatting changed the file contents.");
   }
 
   return absl::OkStatus();
 }
 
+absl::Status RealMain(absl::Span<const std::string_view> input_paths,
+                      bool in_place, bool error_on_changes,
+                      bool opportunistic_postcondition,
+                      const std::string& mode_str) {
+  // Note notable restrictions we place on the CLI, to avoid confusing results /
+  // interactions:
+  //
+  // - If stdin is the input, it should be the only input.
+  // - If we're *not* doing in-place formatting, there should be only one input
+  // (otherwise things will be all jumbled together in stdout),
+  // - If we error-on-changes or use the opportunistic postcondition, there
+  //   should be only one input, to avoid semantic ambiguity when we're
+  //   side-effecting the formatting of files and when we flag the error.
+  bool has_stdin_arg =
+      std::any_of(input_paths.begin(), input_paths.end(),
+                  [](std::string_view path) { return path == "-"; });
+  if (has_stdin_arg) {
```

Comment:
Thoughts on adding tests for each of these cases?

Reply:
I think I've got testing in the .py for the interesting modes and combinations now -- LMK if you see any more that seem like they're missing!